app/controllers/posts_controller.rb

```diff
@@ -1,4 +1,5 @@
 class PostsController < ApplicationController
+  authorize_resource
```

Comment:
- [ ] You should rather use `load_and_authorize` since the Post resource has to be explicitly loaded by you . `load_and_authorize_resource`: This method combines both the loading and authorization of a resource in a controller action. It automatically loads the resource based on the URL parameters or the request context and performs the authorization step.
authorize_resource: This method is used to authorize a specific resource in a controller action. It assumes that the controller action is already loading the resource and only performs the authorization step.
The key difference is that authorize_resource assumes the resource is already loaded, while load_and_authorize_resource takes care of both loading and authorization in one step.